pkgs/servers/sql/cockroachdb/generic.nix

```diff
@@ -0,0 +1,63 @@
+{ lib, stdenv, buildGoModule, fetchurl, cmake, xz, which, autoconf, ncurses6
+, libedit, libunwind, installShellFiles, removeReferencesTo, go, version, sha256
+}:
+
+buildGoModule rec {
+  pname = "cockroach";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://binaries.cockroachdb.com/cockroach-v${version}.src.tgz";
+    inherit sha256;
+  };
+  vendorSha256 = null;
+
+  NIX_CFLAGS_COMPILE = lib.optionals stdenv.cc.isGNU [
+    "-Wno-error=deprecated-copy"
+    "-Wno-error=redundant-move"
+    "-Wno-error=pessimizing-move"
+  ];
+
+  nativeBuildInputs = [ installShellFiles cmake xz which autoconf ];
+  buildInputs = if stdenv.isDarwin then [ libunwind libedit ] else [ ncurses6 ];
+
+  postPatch = ''
+    patchShebangs .
```

Comment:
In https://github.com/NixOS/nixpkgs/pull/124378/files#r730913143 it was recommended to specifically patch only those files in need of patching. Not sure what the advantage is.

Reply:
Running a quick grep on the code base seems to show a decent number of this kind of usage. As well as others that run it on entire directories. If it is really needed, I can go through and change it though.

Reply:
No, let's leave it like this then and go for a quicker, painless merge :)